python/ray/tune/ray_trial_executor.py

```diff
@@ -198,7 +206,7 @@ def _stop_trial(self, trial, error=False, error_msg=None,
             logger.exception("Trial %s: Error stopping runner.", trial)
             self.set_status(trial, Trial.ERROR)
         finally:
-            trial.runner = None
+            trial.set_runner(None)
 
     def start_trial(self, trial, checkpoint=None):
```

Comment:
so to be clear, `start_trial` will now attempt to start the trial multiple times from checkpoint? 

and when does `clear_checkpoint` get hit? wouldn't you just clear out all of your attempts without clearing the checkpoint?

Reply:
the `except Exception` block is just there now as a catch-all for unanticipated errors (i'm not sure if it can even be hit). the reason we don't clear checkpoint in it anymore is because the error couldn't have been caused by a checkpoint, since we don't catch that error there anymore. its caught in `trial_runner` while processing the restore result.

> and when does clear_checkpoint get hit? wouldn't you just clear out all of your attempts without clearing the checkpoint?

this is now in `trial_runner.py#_try_recover`, lmk if its more clear to keep it in `start_trial`.